### ISSUE TITLE
what is a "public room"

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix-doc#2612]
tbh this is not a problem that is limited to the spec, and it probably overlaps with other existing issues, but I'm going to file it here for want of anywhere better right now.

Essentially, the term "public room" (or its counterpart, "private room") gets bandied around (most recently in https://github.com/matrix-org/synapse/pull/7639), but there is no good definition of what defines whether a room is public or private.

For example, is it determined by:
 * the join rule (https://matrix.org/docs/spec/client_server/r0.6.1#m-room-join-rules)
 * the history visibility (https://matrix.org/docs/spec/client_server/r0.6.1#module-history-visibility) (which to be fair doesn't use the words `public` or `private`, but tends to get tied up here)
 * whether the room is included in the results of `publicRooms` (https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-publicrooms)
 * which preset was used to create the room with `/createRoom` (https://matrix.org/docs/spec/client_server/r0.6.1#post-matrix-client-r0-createroom)

**Comment by turt2live** [comment]
fwiw riot-web considers the join rules as the single identifier of public. The history visibility goes a step further to indicate how searchable/on-the-record it is.

In my mind the presets are just templates and no different from someone specifying the state events manually.

**Comment by richvdh** [comment]
another thing that sometimes get confused here is the dm flag.

**Comment by richvdh** [comment]
> fwiw riot-web considers the join rules as the single identifier of public. The history visibility goes a step further to indicate how searchable/on-the-record it is.

I think this is probably fair. We could work towards having that be what we "normally" mean by public/private, and make it explicit when we are using an alternative meaning.

tbh a lot of the problem here is the unfortunate name of `/publicRooms`.

**Comment by richvdh** [comment]
(synapse's config file appears to use `public` to refer to publicRooms :-S )

**Comment by richvdh** [comment]
> In my mind the presets are just templates and no different from someone specifying the state events manually.

the confusing thing here is that, if you don't choose one, the preset is chosen based on the *visibility*, not the join rules. :/

**Comment by turt2live** [comment]
*table flip*

**Comment by anoadragon453** [comment]
After discussions on matrix with @ara4n, @turt2live and @richvdh, we've decided that the presets are a suitable way of describing whether a room is "private" or "public", due to:

* Riot(-web at least) is using the presets when creating a public vs. private rooms
    * Though it's important to note that it later uses the join rules for the lock/globe icons
* Synapse also deciding on a preset to use when creating a room (if one is not provided), based on other parameters provided by the user.

As such, we're going to use presets to differentiate between public/private rooms in https://github.com/matrix-org/synapse/pull/7639.

**Comment by turt2live** [comment]
@anoadragon453 how do you plan to use presets to determine if a room is public after creation?

**Comment by turt2live** [comment]
or when the room becomes "private" under some metric.

**Comment by turt2live** [comment]
oh, I guess for the particular synapse issue it doesn't matter. For notifications it matters.

**Comment by anoadragon453** [comment]
Indeed, my issue was just at creation, not run time. Presets seem to only solve the former.

**Comment by richvdh** [comment]
@anoadragon453 

> After discussions on matrix with @ara4n, @turt2live and @richvdh, we've decided that the presets are a suitable way of describing whether a room is "private" or "public"

I think this is only true in the context of https://github.com/matrix-org/synapse/pull/7639. I think it's still the case that the terms "public" and "private" are unclear in the general case.
